src/txn/txn_ckpt.c

```diff
@@ -1485,6 +1488,32 @@ __checkpoint_lock_dirty_tree(
     return (ret);
 }
 
+/*
+ * __checkpoint_apply_obsolete --
+ *     Returns true if the checkpoint is obsolete.
+ */
+static bool
+__checkpoint_apply_obsolete(WT_SESSION_IMPL *session, WT_BTREE *btree, WT_CKPT *ckpt)
+{
+    if (ckpt->size != 0) {
+        /*
+         * Mark the btree has obsolete page if the checkpoint has valid stop timestamp. This flag is
+         * used to avoid skipping the btree until the obsolete check is performed on the
+         * checkpoints.
+         */
+        if (ckpt->ta.newest_stop_ts != WT_TS_MAX)
+            F_SET(btree, WT_BTREE_OBSOLETE_PAGES);
+        if (__wt_txn_visible_all(session, ckpt->ta.newest_stop_txn,
+              (ckpt->ta.newest_stop_ts == WT_TS_MAX ? WT_TS_MAX :
+                                                      ckpt->ta.newest_stop_durable_ts))) {
```

Comment:
If you used a local variable, you could avoid the strange formatting and make the arg parsing clearer. Since we already have a conditional checking `ckpt->ta.newest_stop_ts`. It could look like:
```
wt_timestamp_t stop_ts;

stop_ts = WT_TS_MAX;
if (ckpt->ta.newest_stop_ts != WT_TS_MAX) {
    F_SET(btree, WT_BTREE_OBSOLETE_PAGES);
    stop_ts = ckpt->ta.newest_stop_durable_ts;
}
if (__wt_txn_visible_all(session, ckpt->ta.newest_stop_txn, stop_ts)) {
...
```

Reply:
Thanks. Fixed as you suggested. It is a very good improvement.